docs/guides/migrating-from-camunda-platform-7/migration-readiness.md

```diff
@@ -0,0 +1,278 @@
+
+2. [The Camunda Platform 7 Adapter](https://github.com/camunda-community-hub/camunda-7-to-8-migration/tree/main/camunda-7-adapter). This is a library providing a worker to hook in Camunda Platform 7-based glue code. For example, it can invoke existing JavaDelegate classes.
+
+3. [A process instance migration tool](https://github.com/camunda-community-hub/camunda-7-to-8-migration/tree/main/process-instance-migration) to migrate running process instances from Camunda Platform 7 to Camunda Platform 8. Ideally, you should let running instances finish prior to migrating.
+
+The tools mentioned above are a good starting point, but are only one option for how you can approach your migration.
+
+## Prepare for smooth migrations
+
+Sometimes you might not be able to use Camunda Platform 8 right away as described in [What to do When You Can’t Quickly Migrate to Camunda Platform 8](https://camunda.com/blog/2022/05/what-to-do-when-you-cant-quickly-migrate-to-camunda-8/). In this case, you will keep developing Camunda Platform 7 process solutions, but you should establish some practices as quickly as possible to ease migration projects later on.
+
+To implement Camunda Platform 7 process solutions that can be easily migrated, stick to the following rules and development practices:
+
+- Implement what we call **Clean Delegates** - concentrate on reading and writing process variables, plus business logic delegation. Data transformations will be mostly done as part of your delegate (and especially not as listeners, as mentioned below). Separate your actual business logic from the delegates and all Camunda APIs. Avoid accessing the BPMN model and invoking Camunda APIs within your delegates.
+- Don’t use listeners or Spring beans in expressions to do data transformations via Java code.
+- Don’t rely on an ACID transaction manager spanning multiple steps or resources.
+- Don’t expose Camunda APIs (REST or Java) to other services or frontend applications.
+- Use primitive variable types or JSON payloads only (no XML or serialized Java objects).
+- Use simple expressions or plug-in FEEL. FEEL is the only supported expression language in Camunda Platform 8. JSONPath is also relatively easy to translate to FEEL. Avoid using special variables in expressions, e.g. `execution` or `task`.
+- Use your own user interface or Camunda Forms; the other form mechanisms are not supported out of the box in Camunda Platform 8.
+- Avoid using any implementation classes from Camunda; generally, those with `\*.impl.\*` in their package name.
+- Avoid using engine plugins.
+
+We also recommend reviewing [BPMN elements supported in Camunda Platform 8](/components/modeler/bpmn/bpmn-coverage.md), though any feature gap will likely be closed soon.
```

Comment:
"though any feature gap will likely be closed soon" -- this is a bit ambiguous, perhaps we could reword in noting "we are actively working on closing feature gaps" or something of the like.